packages/card/src/vaadin-card.js

```diff
@@ -46,7 +299,64 @@ class Card extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
 
   /** @protected */
   render() {
-    return html`<slot></slot>`;
+    return html`
+      <div part="media">
+        <slot name="media"></slot>
+      </div>
+      <div part="header">
+        <slot name="header-prefix"></slot>
+        <slot name="header">
+          <slot name="title"></slot>
+          <slot name="subtitle"></slot>
+        </slot>
+        <slot name="header-suffix"></slot>
+      </div>
+      <div part="content">
+        <slot></slot>
+      </div>
+      <div part="footer">
+        <slot name="footer"></slot>
+      </div>
+    `;
+  }
+
+  /** @private */
+  _onSlotChange() {
+    // Chrome doesn't support `:host(:has())`, so we'll recreate that with custom attributes
+    this.toggleAttribute('has-media', this.querySelector(':scope > [slot="media"]'));
+    this.toggleAttribute('has-header', this.querySelector(':scope > [slot="header"]'));
+    this.toggleAttribute(
+      'has-title',
+      this.querySelector(':scope > [slot="title"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute(
+      'has-subtitle',
+      this.querySelector(':scope > [slot="subtitle"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute('has-header-prefix', this.querySelector(':scope > [slot="header-prefix"]'));
+    this.toggleAttribute('has-header-suffix', this.querySelector(':scope > [slot="header-suffix"]'));
+    this.toggleAttribute('has-content', this.querySelector(':scope > :not([slot])'));
+    this.toggleAttribute('has-footer', this.querySelector(':scope > [slot="footer"]'));
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (!CSS.supports('selector(:host(:has([slot])))')) {
+      if (!this.__boundSlotChangeListener) {
+        this.__boundSlotChangeListener = this._onSlotChange.bind(this);
+      }
+      this.shadowRoot.addEventListener('slotchange', this.__boundSlotChangeListener);
+    }
+  }
+
+  /** @protected */
+  disconnectedCallback() {
+    if (this.__boundSlotChangeListener) {
+      this.shadowRoot.removeEventListener('slotchange', this.__boundSlotChangeListener);
+    }
+    super.disconnectedCallback();
   }
```

Comment:
There shouldn't be a need to remove these local listeners. If we take the wrapper approach suggested above, it could use `<div style="display: content" @slotchange="${this._onSlotChange}">`

Reply:
Oh, okay. I thought this was just best practice to always clean up such listeners. Since the wrapper element could eventually be removed, I think I’d keep this listener on the shadow root.

Reply:
Could add the listener in `createRenderRoot()` instead of `connectedCallback()`

```js
  /**
   * @protected
   * @override
   */
  createRenderRoot() {
    const root = super.createRenderRoot();
    root.addEventListener('slotchange', () => this._onSlotChange());
    return root;
  }
```

Reply:
Thanks. Done.